number_theory/prime.h

```diff
@@ -149,6 +150,41 @@ class EulerSieve {
   }
 };
 
+// Generates all coprime pairs of integers under |num_limit| (inclusive).
+// Each pair (x, y) in the results should satisfy num_limit >= x >= y >= 0.
+template <typename T>
+std::vector<std::pair<T, T>> coprime_pairs(const T &num_limit) {
+  static_assert(std::numeric_limits<T>::is_integer,
+                "coprime_pairs argument |num_limit| must be an integer.");
+  if (num_limit <= 0)
+    return {};
+  uint64_t num_limit_u64 = numeric_cast<uint64_t>(num_limit);
+  std::vector<std::pair<T, T>> pairs;
+
+  // Append pair (x, y) to |pairs| if it is under the limit.
+  // The caller should make sure x >= y.
+  auto add_pair = [&](uint64_t x, uint64_t y) {
```

Comment:
What about capturing just `pairs` instead of everything from the outer scope as reference, i.e., `[&pairs]` instead of `[&]`?